File: epi_judge_java_solutions/epi/MinimumWaitingTime.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.Collections;
import java.util.List;

public class MinimumWaitingTime {
  @EpiTest(testDataFile = "minimum_waiting_time.tsv")

  public static int minimumTotalWaitingTime(List<Integer> serviceTimes) {

    // Sort the service times in increasing order.
    Collections.sort(serviceTimes);

    int totalWaitingTime = 0;
    for (int i = 0; i < serviceTimes.size(); ++i) {
      int numRemainingQueries = serviceTimes.size() - (i + 1);
      totalWaitingTime += serviceTimes.get(i) * numRemainingQueries;
    }
    return totalWaitingTime;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "MinimumWaitingTime.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}

File: epi_judge_cpp_solutions/minimum_waiting_time.cc
#include <algorithm>
#include <iterator>
#include <vector>

#include "test_framework/generic_test.h"

using std::vector;

int MinimumTotalWaitingTime(vector<int> service_times) {
  // Sort the service times in increasing order.
  sort(begin(service_times), end(service_times));

  int total_waiting_time = 0;
  for (int i = 0; i < size(service_times); ++i) {
    int num_remaining_queries = size(service_times) - (i + 1);
    total_waiting_time += service_times[i] * num_remaining_queries;
  }
  return total_waiting_time;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"service_times"};
  return GenericTestMain(args, "minimum_waiting_time.cc",
                         "minimum_waiting_time.tsv", &MinimumTotalWaitingTime,
                         DefaultComparator{}, param_names);
}
